chaoscenter/authentication/pkg/project/repository.go

```diff
@@ -69,51 +69,75 @@ func (r repository) GetProjects(query bson.D) ([]*entities.Project, error) {
 }
 
 // GetProjectsByUserID returns a project based on the userID
-func (r repository) GetProjectsByUserID(userID string, isOwner bool) ([]*entities.Project, error) {
+func (r repository) GetProjectsByUserID(request *entities.ListProjectRequest) (*entities.ListProjectResponse, error) {
 	var projects []*entities.Project
-	query := bson.D{}
 
-	if isOwner == true {
-		query = bson.D{
-			{"members", bson.D{
-				{"$elemMatch", bson.D{
-					{"user_id", userID},
-					{"role", bson.D{
-						{"$eq", entities.RoleOwner},
-					}},
-				}},
-			}}}
-	} else {
-		query = bson.D{
-			{"is_removed", false},
-			{"members", bson.D{
-				{"$elemMatch", bson.D{
-					{"user_id", userID},
-					{"$and", bson.A{
-						bson.D{{"invitation", bson.D{
-							{"$ne", entities.PendingInvitation},
-						}}},
-						bson.D{{"invitation", bson.D{
-							{"$ne", entities.DeclinedInvitation},
-						}}},
-						bson.D{{"invitation", bson.D{
-							{"$ne", entities.ExitedProject},
-						}}},
-					}},
-				}},
-			}}}
+	// Construct the pipeline
+	var pipeline mongo.Pipeline
+
+	pipeline = append(pipeline, utils.CreateMatchStage(request.UserID))
+
+	if request.Filter != nil {
+		filterStages := utils.CreateFilterStages(request.Filter, request.UserID)
+		pipeline = append(pipeline, filterStages...)
+	}
+
+	sortStage := utils.CreateSortStage(request.Sort)
+	if len(sortStage) > 0 {
+		pipeline = append(pipeline, sortStage)
 	}
 
-	result, err := r.Collection.Find(context.TODO(), query)
+	// count stage
+	var pipelineCount mongo.Pipeline
+	pipelineCount = append(pipeline, bson.D{
+		{"$count", "totalNumberOfProjects"},
+	})
```

Comment:
```suggestion
	pipeline = append(pipeline, bson.D{
		{"$count", "totalNumberOfProjects"},
	})
```

Reply:
For this pipeline, you can take ref from: https://github.com/litmuschaos/litmus/blob/master/chaoscenter/graphql/server/pkg/chaos_experiment/handler/handler.go#L508
You can utilise facet stage, which performs multiple operations in parallel. This will help us reduce 1 DB call per request.